loadtest/control/simulcontroller/actions.go

```diff
@@ -1282,3 +1300,357 @@ func sendTypingEventIfEnabled(u user.User, channelId string) error {
+	for i := 0; i < numScrolls; i++ {
+		threads, err = u.GetUserThreads(team.Id, &model.GetUserThreadsOpts{
+			PageSize:   25,
+			Extended:   false,
+			Deleted:    false,
+			Unread:     false,
+			Since:      0,
+			TotalsOnly: false,
+			Before:     oldestThreadId,
+		})
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		oldestThreadId = threads[len(threads)-1].PostId
+		// idle time between scrolls, between 1 and 10 seconds.
+		idleTime := time.Duration(1+rand.Intn(10)) * time.Second
+		select {
+		case <-c.stopChan:
+			return control.UserActionResponse{Info: "action canceled"}
+		case <-time.After(idleTime):
+		}
+	}
+
+	// Switch to "Unread" tabs
+	unreadThreads, err := u.Store().ThreadsSorted(true, false)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if len(unreadThreads) == 0 {
+		unreadThreads, err = u.GetUserThreads(team.Id, &model.GetUserThreadsOpts{
+			PageSize:   25,
+			Extended:   false,
+			Deleted:    false,
+			Unread:     true,
+			Since:      0,
+			TotalsOnly: false,
+		})
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+	}
+
+	oldestUnreadThreadId := unreadThreads[len(unreadThreads)-1].PostId
+	// scrolling between 1 and 3 times
+	numScrolls = rand.Intn(3) + 1
+	for i := 0; i < numScrolls; i++ {
+		unreadThreads, err = u.GetUserThreads(team.Id, &model.GetUserThreadsOpts{
+			PageSize:   25,
+			Extended:   false,
+			Deleted:    false,
+			Unread:     true,
+			Since:      0,
+			TotalsOnly: false,
+			Before:     oldestUnreadThreadId,
+		})
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		oldestUnreadThreadId = unreadThreads[len(unreadThreads)-1].PostId
+		// idle time between scrolls, between 1 and 10 seconds.
+		idleTime := time.Duration(1+rand.Intn(10)) * time.Second
+		select {
+		case <-c.stopChan:
+			return control.UserActionResponse{Info: "action canceled"}
+		case <-time.After(idleTime):
+		}
+	}
+
+	return control.UserActionResponse{Info: "Visited Global Threads Screen"}
+}
+
+func (c *SimulController) followThread(u user.User) control.UserActionResponse {
+	collapsedThreads, resp := control.CollapsedThreadsEnabled(u)
+	if resp.Err != nil || !collapsedThreads {
+		return resp
+	}
+	channel, err := u.Store().CurrentChannel()
+	if err != nil {
+		if errors.Is(err, memstore.ErrChannelNotFound) {
+			return control.UserActionResponse{Info: "followThread: current channel not set"}
+		}
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	post, err := u.Store().RandomPostForChannel(channel.Id)
+	if err != nil {
+		if errors.Is(err, memstore.ErrPostNotFound) {
+			return control.UserActionResponse{Info: "followThread: can't find post to follow in channel"}
+		}
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	try := 0
+	for try < 100 && post.RootId != "" {
```

Comment:
I don't understand why we are repeating this 100 times. Can you expand a bit?

Reply:
We're only looking for a reply post (i.e. an actual thread) so that we can follow that thread. We can theoretically follow a root post that is not yet a thread, but I wanted to replicate user actions more accurately. 
100 times because I didn't want to spin infinitely looking for a random post that is also a reply. 

As @streamer45 said, its better if I made a better method in the store.

Reply:
Can we get a random post, and if it's a root post, reply to it and follow it. Trying something 100 times feels a bit inelegant.